mysys/perf_counters.cc

```diff
@@ -0,0 +1,188 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+#include "perf_counters.h"
+#include <cassert>
+#include <memory>
+
+#if defined(FB_DYNO) && FB_DYNO
+#include <dyno_counters.h>
+#endif
+
+#include <ctime>
+
+#define diff_timespec(TS1, TS2) \
```

Comment:
inline function instead of macro?
```suggestion
static inline uint64_t diff_timespec(const timespec& ts1, const timespec& ts2) noexcept
```